app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -348,17 +355,103 @@ private void setRoots() {
     }
 
     private int initializeAppResources() {
-        //There should be exactly one of these for now
+        // Before we try to initialize a new app, check if any existing apps were left in a
+        // partially deleted state, and finish uninstalling them if so
         for (ApplicationRecord record : getGlobalStorage(ApplicationRecord.class)) {
-            if (record.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
-                //We have an app record ready to go
+            if (record.getStatus() == ApplicationRecord.STATUS_DELETE_REQUESTED) {
+                record.uninstall(this);
+            }
+        }
+        
+        // There may now be multiple app records in storage, because of multiple apps support.
+        // Since this method is called from onCreate of CommCareApplication, current behavior is
+        // just to pick the first app in the list to initialize
+        for(ApplicationRecord record : getGlobalStorage(ApplicationRecord.class)) {
+            if(record.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
                 return initializeAppResources(new CommCareApp(record));

```

Comment:
Your reasoning is exactly right, and I agree with you, but not sure if we have a choice- LoginActivity's onCreate method depends on already knowing what the "current app" is to set up some preferences / display stuff. We could consciously make the decision to stop doing that and have the login screen be app-agnostic, if there are multiple apps installed, until the user makes a choice in the dropdown menu. Would have to look a little closer at LoginActivity to see what exactly would have to change, but it seems reasonable. @ctsims, what do you think?